cmd/object-handlers-common.go

```diff
@@ -264,3 +266,65 @@ func setPutObjHeaders(w http.ResponseWriter, objInfo ObjectInfo, delete bool) {
 		}
 	}
 }
+
+// ilmLimitNoncurrentVersions limits the number of noncurrent versions of an
+// object based on lifecycle rules configured on the bucket.
+func ilmLimitNoncurrentVersions(ctx context.Context, objectAPI ObjectLayer, bucket, object string) (err error) {
+	var lc *lifecycle.Lifecycle
+	if lc, err = globalLifecycleSys.Get(bucket); err != nil {
+		if errors.Is(err, BucketLifecycleNotFound{Bucket: bucket}) {
+			return nil
+		}
+		return err
```

Comment:
Maybe we need log the error and return nil instead.

Reply:
Are you suggesting that we log the "Bucket not found" error too? All errors returned by this function are logged by the caller.